src/main/java/org/joshsim/engine/geometry/GeometryFactory.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Structures describing geometric shapes and their properties.
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.joshsim.engine.geometry;
+
+import java.math.BigDecimal;
+import org.locationtech.spatial4j.context.SpatialContext;
+import org.locationtech.spatial4j.shape.Rectangle;
+import org.locationtech.spatial4j.shape.ShapeFactory;
+
+/**
+ * Factory methods for creating geometric shapes.
+ */
+public class GeometryFactory {
+  private static final ShapeFactory shapeFactory = SpatialContext.GEO.getShapeFactory();
+
+  /**
+   * Creates a square geometry with the specified width and center.
+   *
+   * @param width The width of the square
+   * @param centerLatitude The center latitude
+   * @param centerLongitude The center longitude
+   * @return A Geometry object representing a square
+   */
+  public static Geometry createSquare(
+      BigDecimal width,
```

Comment:
Sorry one more thought popped up. The only other thing I’m not sure about is if this could take Unit so we can specify size (not position) in meters or in degrees. It would be nice if this could handle that even if it is just a haversine. This could be here or it could be in another class that maybe adds that capability. Curious what you think @GondekNP.